### PR TITLE
updated base URL connectivity check

### PR DESCRIPTION
**Pull request by nirmalashish**
- updated base URL connectivity check 
- added timeout as parameter in execCommand method to reduce wait during command process

### CODE REVIEW

File: src/main/java/com/znsio/e2e/tools/Drivers.java [hunk 1]
```diff
@@ -610,9 +608,9 @@ private void attachLogsAndCloseDriver (TestExecutionContext context, String key)
                 break;
             case Driver.APPIUM_DRIVER:
                 if (Runner.platform.equals(Platform.windows)) {
-                    closeAppOnMachine(driver);
+                    closeAppOnMachine(key, driver);
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/Drivers.java:611]: rename key to a more meaningful variable name

File: src/main/java/com/znsio/e2e/tools/Drivers.java [hunk 2]
```diff
@@ -621,58 +619,76 @@ private void attachLogsAndCloseDriver (TestExecutionContext context, String key)
         }
     }
 
-    private void closeWebDriver(String key, Driver driver) {
-        String message = "Browser logs for user: " + key;
+    private void closeWebDriver(String key, @NotNull Driver driver) {
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/Drivers.java:622]: rename key to a more meaningful variable name

File: src/main/java/com/znsio/e2e/tools/Drivers.java [hunk 3]
```diff
@@ -621,58 +619,76 @@ private void attachLogsAndCloseDriver (TestExecutionContext context, String key)
...
-            LOGGER.info(String.format("Closing WebDriver for user: '%s'", key));
+            logMessage = String.format("Closing WebDriver for user: '%s'", key);
+            LOGGER.info(logMessage);
+            ReportPortal.emitLog(logMessage, DEBUG, new Date());
             webDriver.quit();
         }
     }
 
-    private void closeAppOnMachine(Driver driver) {
+    private void closeAppOnMachine(String key, @NotNull Driver driver) {
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/Drivers.java:645]: can this method be renamed to reflect this is for WindowsDriver?

File: src/main/java/com/znsio/e2e/tools/cmd/CommandLineExecutor.java [hunk 4]
```diff
@@ -9,8 +9,13 @@
 
 public class CommandLineExecutor {
     private static final Logger LOGGER = Logger.getLogger(CommandLineExecutor.class.getName());
+    private static final int DEFAULT_COMMAND_TIMEOUT = 120;
```
Comment (anandbagmar) [inline src/main/java/com/znsio/e2e/tools/cmd/CommandLineExecutor.java:12]: Given in many cases you are now passing the timeout as 60, should default itself should be 60?

File: src/main/java/com/znsio/e2e/tools/Drivers.java (hunk 2 above)
Comment (nirmalashish) [reply src/main/java/com/znsio/e2e/tools/Drivers.java]: updated 'key' to 'userPersona'

File: src/main/java/com/znsio/e2e/tools/Drivers.java (hunk 3 above)
Comment (nirmalashish) [reply src/main/java/com/znsio/e2e/tools/Drivers.java]: updated methods for windows and android to more meaningful names

File: src/main/java/com/znsio/e2e/tools/cmd/CommandLineExecutor.java (hunk 4 above)
Comment (nirmalashish) [reply src/main/java/com/znsio/e2e/tools/cmd/CommandLineExecutor.java]: done

**Pull request merged**